pkg/services/rendering/interface.go

```diff
@@ -42,18 +42,25 @@ func getRequestTimeout(opt TimeoutOpts) time.Duration {
 	return opt.Timeout * opt.RequestTimeoutMultiplier
 }
 
-type Opts struct {
+type CommonOpts struct {
```

Comment:
I've extracted common opts. We could use `Opts` instead of having different structs, but I did this as a first step since I want to avoid changing all returning types.